From patchwork Thu Sep 24 06:30:43 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Jian-Hong Pan <jhp@endlessos.org>
X-Patchwork-Id: 11796255
Return-Path: 
 <SRS0=xrLY=DB=lists.infradead.org=linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 7652892C
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Thu, 24 Sep 2020 06:35:46 +0000 (UTC)
Received: from merlin.infradead.org (merlin.infradead.org [205.233.59.134])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.kernel.org (Postfix) with ESMTPS id 446A02067B
	for <patchwork-linux-arm@patchwork.kernel.org>;
 Thu, 24 Sep 2020 06:35:46 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=lists.infradead.org
 header.i=@lists.infradead.org header.b="Ewe+VItP";
	dkim=fail reason="signature verification failed" (2048-bit key)
 header.d=endlessos-org.20150623.gappssmtp.com
 header.i=@endlessos-org.20150623.gappssmtp.com header.b="CNwT7A48"
DMARC-Filter: OpenDMARC Filter v1.3.2 mail.kernel.org 446A02067B
Authentication-Results: mail.kernel.org;
 dmarc=none (p=none dis=none) header.from=endlessos.org
Authentication-Results: mail.kernel.org;
 spf=none
 smtp.mailfrom=linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=merlin.20170209; h=Sender:Content-Transfer-Encoding:
	Content-Type:Cc:List-Subscribe:List-Help:List-Post:List-Archive:
	List-Unsubscribe:List-Id:MIME-Version:Message-Id:Date:Subject:To:From:
	Reply-To:Content-ID:Content-Description:Resent-Date:Resent-From:Resent-Sender
	:Resent-To:Resent-Cc:Resent-Message-ID:In-Reply-To:References:List-Owner;
	bh=chkjbj+LzntWzzQJLh/rXGTFqeExn4dgG2O97gwtks0=; b=Ewe+VItPNb0cS0qCak0A3gLOeX
	bcpHL7ZaEhLIPRz8yVoIAmkgeWa7i/Di2PJYMylsX4RKDFgyUL1h7ZX4z+1ttbXbvrB+18WV8oIJT
	T1eJhGvQryqOaiLYK3c7IDdlJWp7pG4L4hTz/aKyJwesEIRAbBwiWZCG1bzOo55Z02w3hpnC7yJCC
	+9chtonwVExWOlHsIWPQMLY4ONWVnB9wjCfIYhNWkB+6ZsfXKyqox9jQW/L8FXbF7ggXiAr5h5eUQ
	ltSVML0uS0EbGETdS66lseFR0udqSyCw8xV9xg2VE8WwRWWwdR8TIqN+9IrMwPEdw08pCKhaTHPQV
	2jHNXI5Q==;
Received: from localhost ([::1] helo=merlin.infradead.org)
	by merlin.infradead.org with esmtp (Exim 4.92.3 #3 (Red Hat Linux))
	id 1kLKqt-0002qa-Il; Thu, 24 Sep 2020 06:35:35 +0000
Received: from mail-pf1-x443.google.com ([2607:f8b0:4864:20::443])
 by merlin.infradead.org with esmtps (Exim 4.92.3 #3 (Red Hat Linux))
 id 1kLKqp-0002pK-Dt
 for linux-arm-kernel@lists.infradead.org; Thu, 24 Sep 2020 06:35:32 +0000
Received: by mail-pf1-x443.google.com with SMTP id z18so1296203pfg.0
 for <linux-arm-kernel@lists.infradead.org>;
 Wed, 23 Sep 2020 23:35:30 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=endlessos-org.20150623.gappssmtp.com; s=20150623;
 h=from:to:cc:subject:date:message-id:mime-version
 :content-transfer-encoding;
 bh=cMk5EkO4+a7or+6O9dGbx3RFLRF/gjrDA61H05E6FkU=;
 b=CNwT7A48Bv2tL3cgjhSN+QqVUZyOE4oH0z/QPxXWxHlfvRHPpqeNrgMWakuntfufGc
 6KUk9TcIO4PkswHtuWMCf8r0mqTH7vFuT/bBhZeuy+TzYNKUXRoJdaXK4dG2kfEf6gk9
 Rs+I9bobAz5kV/oNMpcu35ByBl0v2BMQslpfYGtbn+BM6zjiNUEgodgm/4r34wcvNwjn
 2UZSNR3oyiG2TOzi14qu9qp3NluDDlwqlAsjDtGiwA1n+eCdk+t0tAvQNAGekqGpvOW0
 AfujOjW5OyTXMvkODohRmQk5rm5/k6ztAB9NzO9IKQlmmQAsxA4HMGEoPuDQUbYpeg8O
 Orjg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
 :content-transfer-encoding;
 bh=cMk5EkO4+a7or+6O9dGbx3RFLRF/gjrDA61H05E6FkU=;
 b=IbmaG6cS3n+4zMcqlr0vhRIs13DTbZLUxntFaAUm+Dr4d2PmTeUro/RBbNG7913CRF
 RvUGF5q0E1FCUitInA5paQ554T4SBLC0vtYBR/2syd688HgW9vgZj9Z/Ml8lMwDc1YOX
 unIRkI0POHYdi3iSJgcGMQGttr4yga1xJte9Nl0wyz99hs+7hkq1t6u+i0ce5Rk0BIOe
 vrpXNo2Z3Ky3N8/C1JYi7LRz5s/YyH8F2h87bJHKq0LZjIrAdsex+TquSHI6Oo+JTP2o
 JNq+2N3VH8cXh4EAThKdRYt2uRSYPmMt5zB6aZ7TdmN/KRrUunUTv3azQ7WYPZ2pb87P
 /Jtw==
X-Gm-Message-State: AOAM532LDC+KonHJFYIJ9EWedqQtA5dzYXv+YV8vaedi0MNndgtXMKqj
 25xaRucf+K4XN3v7IvjkRUWonA==
X-Google-Smtp-Source: 
 ABdhPJxRtZ216lgi7xyFj9DVfZBzPvh8X3PV8a9E299xrYCeK5YYgipOFzUGlybGn6AWVn8DDIuoiw==
X-Received: by 2002:a62:5b85:0:b029:142:2501:34e1 with SMTP id
 p127-20020a625b850000b0290142250134e1mr3072236pfb.58.1600929329001;
 Wed, 23 Sep 2020 23:35:29 -0700 (PDT)
Received: from starnight.localdomain (123-204-46-122.static.seed.net.tw.
 [123.204.46.122])
 by smtp.googlemail.com with ESMTPSA id f4sm1691321pgk.19.2020.09.23.23.35.23
 (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
 Wed, 23 Sep 2020 23:35:28 -0700 (PDT)
From: Jian-Hong Pan <jhp@endlessos.org>
To: Rob Herring <robh+dt@kernel.org>, Heiko Stuebner <heiko@sntech.de>,
 Soeren Moch <smoch@web.de>, Tobias Schramm <t.schramm@manjaro.org>,
 Johan Jonker <jbx6244@gmail.com>,
 Katsuhiro Suzuki <katsuhiro@katsuster.net>,
 Hugh Cole-Baker <sigmaris@gmail.com>, Robin Murphy <robin.murphy@arm.com>
Subject: [PATCH] arm64: dts: rockchip: disable USB type-c DisplayPort
Date: Thu, 24 Sep 2020 14:30:43 +0800
Message-Id: <20200924063042.41545-1-jhp@endlessos.org>
X-Mailer: git-send-email 2.28.0
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20200924_023531_484980_A8C102E9 
X-CRM114-Status: GOOD (  15.78  )
X-Spam-Score: 0.0 (/)
X-Spam-Report: SpamAssassin version 3.4.4 on merlin.infradead.org summary:
 Content analysis details:   (0.0 points)
 pts rule name              description
 ---- ----------------------
 --------------------------------------------------
 -0.0 RCVD_IN_DNSWL_NONE     RBL: Sender listed at https://www.dnswl.org/,
 no trust [2607:f8b0:4864:20:0:0:0:443 listed in]
 [list.dnswl.org]
 0.0 SPF_HELO_NONE          SPF: HELO does not publish an SPF Record
 -0.0 SPF_PASS               SPF: sender matches SPF record
 -0.1 DKIM_VALID Message has at least one valid DKIM or DK signature
 0.1 DKIM_SIGNED            Message has a DKIM or DK signature,
 not necessarily
 valid
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.29
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: 
 <http://lists.infradead.org/mailman/options/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: 
 <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>,
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
Cc: devicetree@vger.kernel.org, Jian-Hong Pan <jhp@endlessos.org>,
 linux-kernel@vger.kernel.org, linux-rockchip@lists.infradead.org,
 linux@endlessm.com, linux-arm-kernel@lists.infradead.org
Sender: "linux-arm-kernel" <linux-arm-kernel-bounces@lists.infradead.org>
Errors-To: 
 linux-arm-kernel-bounces+patchwork-linux-arm=patchwork.kernel.org@lists.infradead.org

The cdn-dp sub driver probes the device failed on PINEBOOK Pro.

kernel: cdn-dp fec00000.dp: [drm:cdn_dp_probe [rockchipdrm]] *ERROR* missing extcon or phy
kernel: cdn-dp: probe of fec00000.dp failed with error -22

Then, the device halts all of the DRM related device jobs. For example,
the operations: vop_component_ops, vop_component_ops and
rockchip_dp_component_ops cannot be bound to corresponding devices. So,
Xorg cannot find the correct DRM device.

The USB type-C DisplayPort does not work for now. So, disable the
DisplayPort node until the type-C phy work has been done.

Link: https://patchwork.kernel.org/patch/11794141/#23639877
Signed-off-by: Jian-Hong Pan <jhp@endlessos.org>
---
 arch/arm64/boot/dts/rockchip/rk3399-pinebook-pro.dts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinebook-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinebook-pro.dts
index 06d48338c836..d624c595c533 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinebook-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinebook-pro.dts
@@ -380,7 +380,7 @@ mains_charger: dc-charger {
 };
 
 &cdn_dp {
-	status = "okay";
+	status = "disabled";
 };
 
 &cpu_b0 {
